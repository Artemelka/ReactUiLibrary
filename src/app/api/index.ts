import { getLabels, getLocales } from './localization';

export const API = {
    localization: {
        getLabels,
        getLocales
    }
};
