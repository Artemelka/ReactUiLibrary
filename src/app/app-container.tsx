import React, { Component } from 'react';
import thunk from 'redux-thunk';
import { createBrowserHistory } from 'history';
import { createStore, applyMiddleware } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import { Provider } from 'react-redux';
import { Router } from 'react-router';
import { routerMiddleware } from 'react-router-redux';
import { reducers } from './reducers';
import { TranslateProvider, translateMiddleware } from '../services/translate';
import { request } from '../services';
import { App } from './index';

const history = createBrowserHistory();
const store = createStore(reducers, composeWithDevTools(
    applyMiddleware(
        routerMiddleware(history),
        translateMiddleware,
        thunk
)));

export class AppContainer extends Component {
    fetchDictionary = () => request('/api/translate');

    render() {
        return (
            <Provider store={store}>
                <TranslateProvider fetchDictionary={this.fetchDictionary}>
                    <Router history={history}>
                        <App />
                    </Router>
                </TranslateProvider>
            </Provider>
        );
    }
}
