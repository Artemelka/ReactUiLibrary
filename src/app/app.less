.App {
    bottom: 0;
    //font-family: 'Fira Sans', sans-serif;
    left: 0;
    overflow: auto;
    position: fixed;
    right: 0;
    top: 0;
}
