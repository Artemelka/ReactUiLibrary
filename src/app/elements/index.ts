export { Anchor } from './anchor/Anchor';
export { Button } from './buttons/Button';
export { ButtonsGroup } from './button-group/ButtonGroup';
export { CardList } from './card-list/CardList';
export { Checkbox, Input, RadioButtons, Select, Textarea } from './inputs';
export { DropDownPanel } from './drop-down-panel/DropDownPanel';
export { DropDownDetails } from './drop-down-panel/DropDownDetails';
export { DropDownSummary } from './drop-down-panel/DropDownSummary';
export { IconModule } from './icon';
export { Modal, ModalSize } from './modal/Modal';
export { Progress } from './progress';
export { Text } from './text';
