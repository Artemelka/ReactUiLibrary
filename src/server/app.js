import express from 'express';
import bodyParser from 'body-parser';
import fileUpload from 'express-fileupload';
import reload from 'express-reload';
import mongoose from 'mongoose';
import { DB_URL, DB_URL_CUT, ServerParams, KEEP_ALIVE_TIMEOUT } from './constants';

const app = express();
const { PORT, HOST } = ServerParams;
const path = __dirname + '/routes/';
const listenCallback = error => error
    ? console.log('something bad happened', error)
    : console.log(`Listening on port ${PORT}!`);

app.use(bodyParser.json());
app.use(bodyParser.text());
app.use(fileUpload());
app.use(express.static('dist'));
app.use('/api', reload(path));

const appServer = app.listen(PORT, HOST, listenCallback);

mongoose.connect(DB_URL, {
    useNewUrlParser: true,
    useUnifiedTopology: true
})
    .then(() => console.log('Connected to DB!'))
    .catch(error => console.error(error));

appServer.keepAliveTimeout = KEEP_ALIVE_TIMEOUT;
