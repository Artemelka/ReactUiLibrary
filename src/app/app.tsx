import * as React from 'react';
import { Route } from 'react-router';
import { TestHomePage, LibraryPage } from './pages';

import './app.less';

export class App extends React.Component<{}, {}> {
    render() {
        return (
            <div className="App">
                <Route path="/" exact component={TestHomePage} />
                {/*<Route path="/" exact component={LibraryPage} />*/}
                {/*<Route path="/library" component={LibraryPage} />*/}
            </div>
        );
    }
}

