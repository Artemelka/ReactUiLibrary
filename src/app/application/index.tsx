import React from 'react';
import { Provider } from 'react-redux';
import { ConnectedRouter } from 'connected-react-router';
import { AppLoader } from 'components';
import { StoreInjectorProvider } from '@artemelka/redux-store-injector';
import { appStore, history } from './redux';
import { AppContainer } from './App';
import { mockAllRequest } from '../api';

export const App = () => {
    if (process.env && process.env.MOCK) {
        mockAllRequest();
    }

    return (
        <Provider store={appStore}>
            <StoreInjectorProvider store={appStore}>
                <ConnectedRouter history={history}>
                    <AppContainer />
                    <AppLoader />
                </ConnectedRouter>
            </StoreInjectorProvider>
        </Provider>
    );
};
