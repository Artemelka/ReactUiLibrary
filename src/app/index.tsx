import React from 'react';
import { Provider } from 'react-redux';
import { ConnectedRouter } from 'connected-react-router';
import { appStore } from './store';
import { history } from './app-history';
import { AppContainer } from './app';
import { AppLoader } from './components';
import { mockAllRequest } from './api';

export const App = () => {
    if (process.env && process.env.MOCK) {
        mockAllRequest();
    }

    return (
        <Provider store={appStore}>
            <ConnectedRouter history={history}>
                <AppContainer />
                <AppLoader />
            </ConnectedRouter>
        </Provider>
    );
};
