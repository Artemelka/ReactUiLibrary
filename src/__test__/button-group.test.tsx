import React from 'react';
import { shallow } from 'enzyme';
import { ButtonGroup, Button } from '../app/elements';

const GROUP_ITEM_MODIFIER = 'Button-group__item--separator-medium';
const testMockFirstButton = jest.fn();
const testMockSecondtButton = jest.fn();
const testMockThirdButton = jest.fn();
const buttonsProps = [
    {
        label: 'test 1',
        onClick: testMockFirstButton
    }, {
        label: 'test 2',
        onClick: testMockSecondtButton
    }, {
        label: 'test 3',
        onClick: testMockThirdButton
    }
];

describe('Test ButtonGroup', () => {
    test('Rendering Snapshot', () => {
        const wrapper = shallow(
            <ButtonGroup.Component>
                {buttonsProps.map((props, index) => <Button key={index} {...props}/>)}
            </ButtonGroup.Component>
        );
        const groupItemCollection = wrapper.find('.Button-group__item');

        expect(wrapper).toMatchSnapshot();
        expect(groupItemCollection.at(1).hasClass(GROUP_ITEM_MODIFIER)).toBe(false);
    });
    test('Rendering Snapshot with separator', () => {
        const wrapper = shallow(
            <ButtonGroup.Component separatorSize={ButtonGroup.SeparatorSize.MEDIUM}>
                {buttonsProps.map((props, index) => <Button key={index} {...props}/>)}
            </ButtonGroup.Component>
        );
        const groupItemCollection = wrapper.find('.Button-group__item');

        expect(wrapper).toMatchSnapshot();
        expect(groupItemCollection.at(0).hasClass(GROUP_ITEM_MODIFIER)).toBe(false);
        expect(groupItemCollection.at(1).hasClass(GROUP_ITEM_MODIFIER)).toBe(true);
    });
    test('expect round style in Button with prop round', () => {
        const wrapper = shallow(
            <ButtonGroup.Component round>
                {buttonsProps.map((props, index) => <Button key={index} {...props}/>)}
            </ButtonGroup.Component>
        );
        const buttonCollection = wrapper.find('Button');

        expect(buttonCollection.at(0).prop('roundLeft')).toBe(true);
        expect(buttonCollection.at(1).prop('roundLeft')).toBe(false);
        expect(buttonCollection.at(1).prop('roundRight')).toBe(false);
        expect(buttonCollection.at(2).prop('roundRight')).toBe(true);
    });
});
