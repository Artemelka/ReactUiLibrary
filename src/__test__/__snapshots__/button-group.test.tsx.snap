// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test ButtonGroup Rendering Snapshot 1`] = `
<div
  className="Button-group"
>
  <Button
    label="test 1"
    onClick={[MockFunction]}
    type="button"
  />
  <span
    className="Button-group__separator Button-group__separator--small"
  />
  <Button
    label="test 2"
    onClick={[MockFunction]}
    type="button"
  />
  <span
    className="Button-group__separator Button-group__separator--small"
  />
  <Button
    label="test 2"
    onClick={[MockFunction]}
    type="button"
  />
</div>
`;

exports[`Test ButtonGroup Rendering Snapshot with separator 1`] = `
<div
  className="Button-group"
>
  <Button
    label="test 1"
    onClick={[MockFunction]}
    type="button"
  />
  <span
    className="Button-group__separator Button-group__separator--medium"
  />
  <Button
    label="test 2"
    onClick={[MockFunction]}
    type="button"
  />
  <span
    className="Button-group__separator Button-group__separator--medium"
  />
  <Button
    label="test 2"
    onClick={[MockFunction]}
    type="button"
  />
</div>
`;
